pennylane/math/matrix_manipulation.py

```diff
@@ -224,35 +210,51 @@ def _sparse_expand_matrix(base_matrix, wires, wire_order, format="csr"):
     Returns:
         tensor_like: expanded matrix
     """
-    n_wires = len(wires)
+    base_matrix.eliminate_zeros()
+
     n_total_wires = len(wire_order)
 
-    if isinstance(wires, qml.wires.Wires):
-        expanded_wires = wires.tolist()
-    else:
-        expanded_wires = list(copy.copy(wires))
+    wires = wires.tolist() if isinstance(wires, qml.wires.Wires) else list(copy.copy(wires))
 
+    expanded_wires = []
+    mats = []
+    op_wires_in_list = False
+    i_count = 0
     for wire in wire_order:
         if wire not in wires:
+            i_count += 1
             expanded_wires.append(wire)
-
-    num_missing_wires = n_total_wires - n_wires
-    if num_missing_wires > 0:
-        expanded_matrix = kron(
-            base_matrix, eye(2**num_missing_wires, format=format), format=format
-        )  # added missing wires at the end
+        elif not op_wires_in_list:
+            if i_count > 0:
+                mats.append(eye(2**i_count, format="coo"))
+            i_count = 0
+            mats.append(base_matrix)
+            op_wires_in_list = True
+            expanded_wires.extend(wires)
+
+    if i_count > 0:
+        mats.append(eye(2**i_count, format="coo"))
+
+    if len(mats) > 1:
+        expanded_matrix = reduce(lambda i, j: kron(i, j, format="coo"), mats)
     else:
         expanded_matrix = copy.copy(base_matrix)
 
-    U = eye(2**n_total_wires, format=format)
+    expanded_matrix.eliminate_zeros()
+
+    U = None
     for i in range(n_total_wires):
         if expanded_wires[i] != wire_order[i]:
+            if U is None:
+                U = eye(2**n_total_wires, format=format)
             j = expanded_wires.index(wire_order[i])  # location of correct wire
-            U = U @ _sparse_swap_mat(
-                i, j, n_total_wires, format=format
-            )  # swap incorrect wire for correct wire
+            U = U @ _sparse_swap_mat(i, j, n_total_wires)  # swap incorrect wire for correct wire
+            U.eliminate_zeros()
 
             expanded_wires[i], expanded_wires[j] = expanded_wires[j], expanded_wires[i]
-
-    expanded_matrix = U.T @ expanded_matrix @ U
-    return expanded_matrix.asformat(format)
+    if U is not None:
+        expanded_matrix = U.T @ expanded_matrix @ U
+        expanded_matrix.eliminate_zeros()
+    if format != "coo":
+        expanded_matrix = expanded_matrix.asformat(format)
+    return expanded_matrix
```

Comment:
```suggestion
    return expanded_matrix.asformat(format)
```
It is a no-op for Scipy to convert to a representation already in place.
Given that you can just ask to convert to the format specified.
I don't think this will be necessary if you set all terms in the matrix product to be of the asked format.

On the other hand, for performance purposes, you can convert all terms participating in the matrix product internally to CSR, and convert to the desired form at the end.

Reply:
Done!